packages/runtime/src/components/builtin/Text/EditableText/useSyncDOMSelection.tsx

```diff
@@ -0,0 +1,41 @@
+import { ReactEditor } from 'slate-react'
+import { useIsomorphicLayoutEffect } from '../../../hooks/useIsomorphicLayoutEffect'
+import { Editor, Range as SlateRange } from 'slate'
+
+/**
+ * Clicking outside of the host blurs our `<Editable />`.
+ * `<Editable />` only updates the DOM's selection to match slate when it is focused.
+ * In the case of a panel being clicked this hook updates the DOM selection to match slate. 
+ */
+export function useSyncDOMSelection(editor: Editor) {
+  useIsomorphicLayoutEffect(() => {
```

Comment:
They use an effect in Slate's `<Editable />` so I decided to do the same. 

Where as before with the `setTimeout` I was sometimes losing the selection on rapid changes, I haven't seen anything like that with `useIsomorphicLayoutEffect`